pkgs/applications/radio/sdrplay/default.nix

```
@@ -0,0 +1,46 @@
+{ stdenv, lib, fetchurl, autoPatchelfHook, libudev }:
+let
+  arch = if stdenv.isx86_64  then "x86_64"
+    else if stdenv.isi686    then "i686"
+    else throw "unsupported architecture";
+in
+stdenv.mkDerivation rec {
+  name = "sdrplay";
+  version = "3.07.1";
+
+  # soapysdrplay also needs to full so name
+  soName = "libsdrplay_api.so." + lib.concatStringsSep "." (lib.take 2 (builtins.splitVersion version));
+
+  src = fetchurl {
+    url = "https://www.sdrplay.com/software/SDRplay_RSP_API-Linux-${version}.run";
+    sha256 = "1a25c7rsdkcjxr7ffvx2lwj7fxdbslg9qhr8ghaq1r53rcrqgzmf";
+  };
+
+  nativeBuildInputs = [ autoPatchelfHook ];
+
+  buildInputs = [ libudev stdenv.cc.cc.lib ];
+
+  unpackPhase = ''
+    sh $src --noexec --target source
+  '';
+
+  sourceRoot = "source";
+
+  buildPhase = "";
```

```suggestion
  dontBuild = true;
```
(I think that should work and it is cleaner)

---

Yep. Looks good and cleaner. Thanks for the hint.